### ISSUE TITLE
Nav sidebar: improve focus management on toggle of sidebar

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#39801]
> I'd expect focus to be transferred to the sidebar when that button is clicked, as well as from the "Navigation Menus" button in the header

See https://github.com/WordPress/gutenberg/pull/39290/#issuecomment-1078648412.

As suggested there let's improve the focus management to transfer to/from the sidebar.

**Comment by getdave** [comment]
Ok I had a look at this. This is a problem for all the secondary sidebars because it isn't handled in the `@wordpress/interface` package. I believe the reason for this is that it's very difficult to have a one-size-fits-all for choosing which element to focus.

I'll keep working on this.

**Comment by alexstine** [comment]
@getdave How does it work currently for block settings sidebar? E.g. using Tab key out of a block will focus block settings sidebar.

**Comment by getdave** [comment]
It's using some hooks to handle going to new focus and back to old one. The complexity in this case is that this sidebar makes use of `ComplementaryArea` from `@wordpress/interface` which is deeply nested and uses a lot of Slot/Fill. This makes auto managing focus quite complex.

**Comment by alexstine** [comment]
@getdave Probably need to use something more flexible. I've faced issues with that myself in another PR.

**Comment by alexstine** [comment]
@getdave Can I help in any way? We've got to land this one before this goes out in 6.0. Labeling issues we can live with, focus management is a must.

**Comment by getdave** [comment]
> @getdave Can I help in any way? We've got to land this one before this goes out in 6.0. Labeling issues we can live with, focus management is a must.

@alexstine I'm painfully aware of this. I've been reaching out looking for additional support as I've been swamped with tasks. I'll circle back to this tomorrow to see if I can find a simple fix with a view to providing something more comprehensive in a future release.

**Comment by scruffian** [comment]
Given that this doesn't happen for the other sidebars, is it a blocker for 6.0?

**Comment by alexstine** [comment]
@scruffian 

> Given that this doesn't happen for the other sidebars, is it a blocker for 6.0?

Absolutely. There is no way this should have happened for other sidebars.

**Comment by getdave** [comment]
This is still valid. Note the sidebar is still experimental so it won't end up in WP 6.1.
